client/keys/add.go

```diff
@@ -45,24 +57,30 @@ phrase, otherwise, a new key will be generated.`,
 	return cmd
 }
 
-// TODO remove the above when addressing #1446
+/*
+input
+	- bip39 mnemonic
+	- bip39 passphrase
+	- bip44 path
+	- local encryption password
+output
+	- armor encrypted private key (saved to file)
+*/
 func runAddCmd(cmd *cobra.Command, args []string) error {
-	var kb keys.Keybase
-	var err error
-	var name, pass string
+	var (
+		kb                   keys.Keybase
+		err                  error
+		name, pass, mnemonic string
+	)
```

Comment:
@rigelrozanski this isn't necessarily the de-facto standard afaik. For example, I personally find this syntax and grouping much easier to read. Regarding the two extra lines, it's really not a big deal given that Golang is already a verbose language. On the other hand, if it makes more sense to move these variables elsewhere that is logically better/closer to their use, then I'm all for it (hinting at your last suggestion).

Ultimately, I think it should be up to the developer's discretion and preference.

Reply:
In all honesty, I don't have a strong preference here. Neither bothers me.

Reply:
>  if it makes more sense to move these variables elsewhere that is logically better/closer to their use, then I'm all for it (hinting at your last suggestion).

Yeah let's do that here because we can here 
________
But yeah, I mean I'm kinda a formatting totalitarian, and think that we should come to consensus on a common set of formatting standards (through reason and rationality - I think we can transcend preference here) and apply it uniformly throughout the SDK - I've outlined my formatting thesis in this case, and plan to keep on developing my thinking around code aesthetics and voicing it - once sufficiently developed it should be explicitly included as linting checks I believe.

Reply:
Amended.

Reply:
@rigelrozanski hmmm, at this point it's worth moving this generalized discussion to another communication channel. 

We are equally formatting totalitarians with drastically different views on a variety of topics. This obviously suggests that a bulk of this discussion would be completely subjective (e.g the formatting thesis you laid out, apart from two extra lines, is imho completely subjective and preferential). I think there is a command ground we can approach for "general guidelines", but a strict de-facto rulebook is not necessarily optimal I think...some ppl are indifferent and the ones that aren't may have drastically different views. I think you even posted a link in slack that suggested that this kind of stuff is futile to adopt.

Happy coding ❤️

Reply:
Well said @alexanderbez, Agree wholeheartedly.

Reply:
tl;dr: we shouldn't block PRs on this as it's mostly subjective but we should discuss and formulate "general guidelines"